react-app/src/component/ColorSelector/ColorSelectorComponent.tsx

```diff
@@ -111,7 +111,7 @@ export const ColorSelectorComponent: React.FC<legendProps> = ({
         ? colorScaleData({ colorsObject, legendColorName }, false)
         : null;
     }
-  };
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
You should add the required dependencies here, otherwise it will trigger each render.